tests/test_merge_dynamic_streams_block.py

```diff
@@ -0,0 +1,163 @@
+from collections import defaultdict
+from time import sleep
+from unittest.mock import MagicMock
+
+from nio.block.terminals import DEFAULT_TERMINAL
+from nio.signal.base import Signal
+from nio.testing.block_test_case import NIOBlockTestCase
+
+from ..merge_dynamic_streams_block import MergeDynamicStreams
+
+
+class TestMergeDynamicStreams(NIOBlockTestCase):
+
+    def process_test_signals(self, blk):
+        blk.process_signals([Signal({"A": "a", "stream": 1})])
+        sleep(0.2)
```

Comment:
One thing I've been doing to get rid of sleeps is mocking the Job class, then triggering the target if we expect it to. So this sleep would go to something like:
```python
with patch('{}.Job'.format(MergeDynamicStreams.__module__.__name__)) as job:
    blk.process_signals(...)
    job.assert_called_once_with(...)
    blk._signal_expiration_job()
```

I realize this sleep will be a little harder to make that happen with, since sometimes we want the job to trigger and sometimes we don't. But in general I think we should try to avoid sleeping in tests.